### PR TITLE
PHP: Try to support php 7.

### LINKED ISSUE #1535: PHP grammar is very outdated
We are closing in to release of PHP 7.4 and soon PHP 8 with JIT
The grammar here only supports upto PHP 5
Also PHP short tags have been removed in 7.4, the grammar should become even shorter.

### PR DESCRIPTION
**Pull request by studentmain**
Should resolve #1535

### CODE REVIEW

File: php/PhpParser.g4 [hunk 1]
```diff
@@ -673,8 +705,9 @@ functionCallName
     : qualifiedNamespaceName
     | classConstant
     | chainBase
+    | parenthesis // TODO: Hack for IIFE, seen in generatorReturn.php
```
Comment (KvanTTT) [inline php/PhpParser.g4]: Maybe `parentheses` (plural form)?

File: php/PhpParser.g4 [hunk 2]
```diff
@@ -489,7 +492,7 @@ expression
     | (Include | IncludeOnce) expression                       #SpecialWordExpression
     | (Require | RequireOnce) expression                       #SpecialWordExpression
 
-    | Static? Function '&'? '(' formalParameterList ')' lambdaFunctionUseVars? blockStatement  #LambdaFunctionExpression
+    | lambdaFunctionExpr                                       #LambdaFunctionExpression
```
Comment (KvanTTT) [inline php/PhpParser.g4]: Please align spaces with other alternative labels

File: php/PhpParser.g4 [hunk 3]
```diff
@@ -539,6 +548,7 @@ assignmentOperator
 
 yieldExpression
     : Yield expression ('=>' expression)?
+    | Yield From functionCall
```
Comment (KvanTTT) [inline php/PhpParser.g4]: Maybe simplify?

```antlr
Yield (expression ('=>' expression)? | From functionCall)
```

File: php/PhpParser.g4 [hunk 4]
```diff
@@ -578,7 +596,14 @@ qualifiedNamespaceName
     ;
 
 namespaceNameList
-    : identifier ('\\' identifier)*
+    : identifier
+    | identifier ('\\' identifier)* '\\' namespaceNameTail
```
Comment (KvanTTT) [inline php/PhpParser.g4]: Also, maybe use an optional operator?

```antlr
| identifier ('\\' identifier)* ('\\' namespaceNameTail)?
```

File: php/PhpParser.g4 [hunk 5]
```diff
@@ -578,7 +596,14 @@ qualifiedNamespaceName
...
+    ;
+
+namespaceNameTail
+    : identifier As identifier
+    | OpenCurlyBracket namespaceNameTail (','namespaceNameTail)* '}'
+    | identifier
```
Comment (KvanTTT) [inline php/PhpParser.g4]:
```antlr
identifier (As identifier)?
```

File: php/PhpParser.g4 (hunk 1 above)
Comment (studentmain) [reply php/PhpParser.g4]: Needs more invesgation, will do it after I finish \u escape syntax.

File: php/PhpParser.g4 (hunk 3 above)
Comment (KvanTTT) [reply php/PhpParser.g4]: I don't think it's necessary because they can be distinguished anyway.

File: php/PhpParser.g4 (hunk 1 above)
Comment (studentmain) [reply php/PhpParser.g4]: Got it. I'll do it tomorrow.

**Comment by evnix** [comment]
> cc: @evnix

Thank you so much!!!!!!
I wish I could help, I am still learning to write these grammars.
Here is the one used by PHP-Parser https://github.com/nikic/PHP-Parser/blob/master/grammar/php7.y but it is YACC/Bison based.

File: php/PhpParser.g4 (hunk 2 above)
Comment (studentmain) [reply php/PhpParser.g4]: should aligned now

**Comment by KvanTTT** [comment]
Is it ready to merge?

**Comment by studentmain** [comment]
Ok, finished it now.

**Comment by KvanTTT** [comment]
Thanks a lot again!

**Comment by evnix** [comment]
@studentmain You are the best!!!

**Pull request merged**